pkg/apis/vsphereprovider/v1alpha1/vsphereproviderstatus_types.go

```diff
@@ -1,33 +1,63 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// VSphereMachineProviderConditionType is a valid value for VSphereMachineProviderCondition.Type
+type VSphereMachineProviderConditionType string
+
+// Valid conditions for an vSphere machine instance
+const (
+	// MachineCreation indicates whether the machine has been created or not. If not,
+	// it should include a reason and message for the failure.
+	MachineCreation VSphereMachineProviderConditionType = "MachineCreation"
+)
+
+// VSphereMachineProviderCondition is a condition in a VSphereMachineProviderStatus
+type VSphereMachineProviderCondition struct {
+	// Type is the type of the condition.
+	Type VSphereMachineProviderConditionType `json:"type"`
+	// Status is the status of the condition.
+	Status corev1.ConditionStatus `json:"status"`
+	// LastProbeTime is the last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
+	// LastTransitionTime is the last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// Reason is a unique, one-word, CamelCase reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
```

Comment:
Semi tempted to say the `Reason` should be a typed string `VSphereMachineProviderConditionReason` and then bring the consts into this package similar to the `Type` and it's available values, WDYT?

Reply:
👍